jdaviz/configs/default/plugins/markers/markers.py

```diff
@@ -0,0 +1,152 @@
+            # 1d spectrum viewers
+            headers += ['index']
+        if self.config in ['specviz2d', 'mosviz']:
+            # 2d spectrum viewers
+            headers += []
+
+        headers += ['data_label']
+
+        self.table.headers_avail = headers
+        self.table.headers_visible = headers
+
+        # subscribe to mouse events on any new viewers
+        self.hub.subscribe(self, ViewerAddedMessage, handler=self._on_viewer_added)
+
+    def _create_viewer_callbacks(self, viewer):
+        if not self.plugin_opened:
+            return
+
+        callback = self._viewer_callback(viewer, self._on_viewer_key_event)
+        viewer.add_event_callback(callback, events=['keydown'])
+
+    def _on_viewer_added(self, msg):
+        self._create_viewer_callbacks(self.app.get_viewer_by_id(msg.viewer_id))
+
+    def _get_mark(self, viewer):
+        matches = [mark for mark in viewer.figure.marks if isinstance(mark, MarkersMark)]
+        if len(matches):
+            return matches[0]
+        mark = MarkersMark(viewer)
+        viewer.figure.marks = viewer.figure.marks + [mark]
+        return mark
+
+    @property
+    def marks(self):
+        return {viewer_id: self._get_mark(viewer)
+                for viewer_id, viewer in self.app._viewer_store.items()
+                if hasattr(viewer, 'figure')}
+
+    @observe('plugin_opened')
+    def _on_plugin_opened_changed(self, *args):
+        if self.disabled_msg:
+            return
+
+        # toggle visibility of markers
+        for mark in self.marks.values():
+            mark.visible = self.plugin_opened
+
+        # subscribe/unsubscribe to keypress events across all viewers
+        for viewer in self.app._viewer_store.values():
+            if not hasattr(viewer, 'figure'):
+                # table viewer, etc
+                continue
+            callback = self._viewer_callback(viewer, self._on_viewer_key_event)
+
+            if self.plugin_opened:
+                viewer.add_event_callback(callback, events=['keydown'])
+            else:
+                viewer.remove_event_callback(callback)
+
+    def _on_viewer_key_event(self, viewer, data):
+        if data['event'] == 'keydown' and data['key'] in ('m', 'M'):
+            # TODO: refactor to share code with mouseover display if PR#1976 merged
+            # TODO: merge with mouseover display entirely and show mouseover info in table
+
+            row_info = self.app.session.application._tools['g-coords-info'].as_dict()
+
+            if 'viewer' in self.table.headers_avail:
+                row_info['viewer'] = viewer.reference if viewer.reference is not None else viewer.reference_id  # noqa
+
+            for k in self.table.headers_avail:
+                row_info.setdefault(k, _default_table_values.get(k, ''))
+
+            try:
+                self.table.add_item(row_info)
+            except ValueError:
+                raise ValueError(f'failed to add {row_info} to table')

```

Comment:
Would help the devs more if you also include the original error.

```suggestion
            except ValueError as err:
                raise ValueError(f'failed to add {row_info} to table: {repr(err)}')
```